solutions/swb-reference/integration-tests/tests/isolated/aws-accounts/update.test.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { resourceTypeToKey } from '@aws/workbench-core-base';
+import _ from 'lodash';
+import ClientSession from '../../../support/clientSession';
+import { AccountHelper } from '../../../support/complex/accountHelper';
+import Account from '../../../support/resources/accounts/account';
+import Setup from '../../../support/setup';
+import HttpError from '../../../support/utils/HttpError';
+import RandomTextGenerator from '../../../support/utils/randomTextGenerator';
+import { checkHttpError } from '../../../support/utils/utilities';
+
+describe('awsAccounts update negative tests', () => {
+  const setup: Setup = new Setup();
+  let adminSession: ClientSession;
+
+  beforeEach(() => {
+    expect.hasAssertions();
+  });
+
+  beforeAll(async () => {
+    adminSession = await setup.getDefaultAdminSession();
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('when updating an account', () => {
+    let account: Account;
+
+    beforeEach(() => {
+      const accountId = `${resourceTypeToKey.account.toLowerCase()}-00000000-0000-0000-0000-000000000000`;
+      account = adminSession.resources.accounts.account(accountId);
+    });
+
+    describe('and the update params are invalid', () => {
+      test('it throws a validation error', async () => {
+        try {
+          const badValue = 1 as unknown as string;
```

Comment:
dark magic 😸 . Could we pass object as unknown directly?
```
await account.update({
            name: 1,
            awsAccountId: 1,
            envMgmtRoleArn: 1,
            hostingAccountHandlerRoleArn: 1,
            externalId: 1
          } as unknown)
```

Reply:
can't do that unfortunately. type system refuses to take it since the method has a known type